rad/prelude/patterns.rad

```diff
@@ -227,3 +228,29 @@
           [1 (/member [1 3 5])] :no
           [1 (/member [2 4 8])] :yes
           _                     :no) ==> :yes])
+
+(def /number
+  "Matches values which are numbers. Matches are passed to the inner pattern."
+  (fn [pat]
+    (fn [v]
+      (if (number? v)
+        (match-pat pat v)
```

Comment:
Interesting pattern (this passing into inner patterns). I see the point of it for the usage above, and am happy with it, but I'm also curious about what the more general principle is (e.g., if the *pattern* is just the number 7, for example, it shouldn't bind an x; intuitively that might mean "any predicate that has more than a one-element extension", but I'm not sure)

Reply:
If the inner pat is `7`, that is `(/number 7)`, then it matches `7`. Any other number would match the outer pattern but then fail at the pattern `7`.

I'm not sure what you mean by "bind an `x`" -- there is no `x` here.